### PR TITLE
[Tested - Review / Feedback Needed] Tor node patch

### LINKED ISSUE #109: lnd with nat enabled crashes
Most probably related to miniupnp or nat-pmp missing in path

```
Scanning local network for a NAT-PMP enabled device
[ERR] SRVR: Unable to discover a NAT-PMP enabled device on the local network: fork/exec /usr/bin/ip: no such file or directory
[ERR] SRVR: unable to create server: Unable to discover a NAT-PMP enabled device on the local network: fork/exec /usr/bin/ip: no such file or directory
```

### PR DESCRIPTION
**Pull request by nolim1t**
Testing out alternative configuration for tor LND nodes.

This may also fix #109 as well

### CODE REVIEW

**Comment by AnotherDroog** [comment]
Letting `lnd` container use host networking is a good idea. That also helps with `UPnP/NAT`.

Does `invoicer` also need to use host networking tho?

**Comment by nolim1t** [comment]
yes it does too. Perhaps we should move it all to host networking so we have a single config?

I don't see too many drawbacks right now.

**Comment by nolim1t** [comment]
I'll put in issue #188 too as this seems quite important

**Comment by AnotherDroog** [comment]
Is this confirmed working?

**Comment by nolim1t** [comment]
This is working. I just merged in some commits from master with this branch so its up to date.

**Comment by nolim1t** [comment]
Still testing this with a fresh seed

**Comment by nolim1t** [comment]
Works however theres a side effect of the tor uri not getting advertised even though its generated.

Generation is done after the node is synced

**Comment by nolim1t** [comment]
appears URIs don't actually get populated (refer to https://github.com/lightningnetwork/lnd/issues/2752) after the onion private key is generated, until a restart is done

**Comment by nolim1t** [comment]
Currently seeing if that bug can be recreated.

```bash
docker exec compose_lnd_1 lncli stop ; docker stop compose_l
nd_1 ; sleep 10 ; rm /media/important/important/lnd/tls* ; rm -fr /media/importa
nt/important/lnd/data/chain ; docker stop compose_invoicer_1 ; docker start comp
ose_lnd_1 ; sleep 5 ; /usr/local/sbin/lncm-createwallet.py ; sleep 5 ; docker st
art compose_invoicer_1
```

**Comment by nolim1t** [comment]
Issue  lightningnetwork/lnd#2752 doesn't apply. The file does get created.

Currently changed to make TOR nodes by default

**Comment by nolim1t** [comment]
testing out now

**Comment by nolim1t** [comment]
I've done a full install using ```./make_img.sh``` and this is working @AnotherDroog .

If noone has any objections for having tor as a default, would like to have this in master

**Comment by nolim1t** [comment]
I'll keep the review period open for a little. No need to rush things yet

**Comment by nolim1t** [comment]
Eventually I'd like to have bitcoind access tor control port too, but lets keep the scope lnd for now

File: home/lncm/compose/tor/docker-compose.yml [hunk 1]
```diff
@@ -5,16 +5,16 @@ services:
     volumes:
       - /media/archive/archive/bitcoin/:/root/.bitcoin/
     ports:
-      - "8332"
-      - "8333"
-      - "28332"
-      - "28333"
+      - "8332:8332"
```
[the remark this answers is not in the document]
Comment (nolim1t) [reply home/lncm/compose/tor/docker-compose.yml]: I removed it

File: home/lncm/compose/clearnet/docker-compose.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ services:
         ipv4_address: 172.16.88.2
 
   lnd:
-    image: lncm/lnd:0.5.2
+    image: lncm/lnd:0.5.2-linux-arm
```
[the remark this answers is not in the document]
Comment (nolim1t) [reply home/lncm/compose/clearnet/docker-compose.yml]: because 0.5.2 is broken

[the remark this answers is not in the document]
Comment (nolim1t) [reply home/lncm/compose/clearnet/docker-compose.yml]: 0.5.2 is broken

**Pull request merged**